packages/roosterjs-editor-api/lib/format/toggleTagCore.ts

```diff
@@ -0,0 +1,88 @@
+import {
+    Browser,
+    Position,
+    wrap,
+    fromHtml,
+    getTagOfNode,
+    splitBalancedNodeRange,
+} from 'roosterjs-editor-dom';
+import { ChangeSource, PositionType, QueryScope } from 'roosterjs-editor-types';
+import { Editor } from 'roosterjs-editor-core';
+
+const ZERO_WIDTH_SPACE = '&#8203;';
+const UNWRAPPABLE_NODES = 'LI,THEAD,TBODY,TR,TD,TH'.split(',');
+
+/**
+ * Toggle a tag at selection, if selection already contains elements of such tag,
+ * the elements will be untagge and other elements will take no affect
+ * @param editor The editor instance
+ * @param tag The tag name
+ * @param wrapFunction The wrap function
+ * @param unwrapFunction The unwrap function
+ * @param styler The styler for setting the style for the blockquote element
+ */
+export default function toggleTagCore<T extends keyof HTMLElementTagNameMap>(
+    editor: Editor,
+    wrapper: T,
+    wrapFunction: (nodes: Node[]) => HTMLElement,
+    unwrapFunction: (node: Node) => Node,
```

Comment:
Can we make wrapFunction and unwrapFunction optional and default behavior is to wrap/unwrap with the given tag since we already know the tag name?

And move styler param before these two, so that toggleBlockQuote doesn't need to pass in these params.

Reply:
sure